samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,177 @@
+
+Run types and data sizes
+------------------------
+
+CoreMark has the following predefined run types and data sizes that are used for data algorithms:
+
+* The :kconfig:option:`CONFIG_COREMARK_RUN_TYPE_PERFORMANCE` Kconfig option - Sets :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` to 2000 bytes.
+* The :kconfig:option:`CONFIG_COREMARK_RUN_TYPE_PROFILE` Kconfig option - Sets :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` to 1200 bytes.
+* The :kconfig:option:`CONFIG_COREMARK_RUN_TYPE_VALIDATION` Kconfig option - Sets :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` to 500 bytes.
+
+You can also specify a custom :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` value and submit your results by following the rules from the `CoreMark GitHub`_ repository.
+Make sure that when setting the :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` Kconfig option the required memory space is available.
+
+Memory allocation methods
+-------------------------
+
+You can select the following memory allocation methods:
+
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` Kconfig option uses the main thread stack for data allocation and is enabled by default.
+  You can change the main stack size by setting the :kconfig:option:`CONFIG_MAIN_STACK_SIZE` Kconfig option.
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` Kconfig option uses Zephyr :c:func:`k_malloc` or :c:func:`k_free` functions to allocate the memory in the heap.
+  You can adjust the heap size using the :kconfig:option:`CONFIG_HEAP_MEM_POOL_SIZE` Kconfig option.
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STATIC` option allows data allocation in the RAM as a regular static variable.
+  In this case, you do not need to be aware of the main thread stack size or heap memory pool size.
+
+Multithread mode
+----------------
+
+CoreMark can also be executed in multithread mode.
+To specify a number of threads, use the :kconfig:option:`CONFIG_COREMARK_THREADS_NUMBER` Kconfig option.
+In multithread mode, CoreMark is executed in all threads simultaneously.
+Each thread runs number of iterations equal to :kconfig:option:`CONFIG_COREMARK_ITERATIONS`.
+By default, the :kconfig:option:`CONFIG_TIMESLICE_SIZE` Kconfig option is set to 10 ms, which imitates RTOS multithread usage.
+However, in the final report, the thread execution is combined.
+In the multithread mode, the :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` Kconfig option is allocated for each thread separately.
+Only the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` and :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` memory methods can be used with multiple threads.
+In case of the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` Kconfig option, the data for all threads is allocated to the main thread stack.
+
+Configuration options
+=====================
+
+Check and configure the following Kconfig options:
+
+.. _CONFIG_APP_MODE_FLASH_AND_RUN:
```

Comment:
Add an entry regarding the CONFIG_SB_APP_CPUNET_RUN. Try to describe it generally e.g. "Enable the benchmark execution also for the Network Core for targets with nRF53 SoC.